x-pack/legacy/plugins/uptime/public/components/functional/filter_group/parse_filter_map.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+interface FilterField {
+  name: string;
+  fieldName: string;
+}
+
+export const parseFiltersMap = (
+  filterMapString: string,
+  fields: FilterField[]
+): Record<string, any[]> => {
+  const filterSlices: Record<string, any[]> = {};
+  try {
+    const map = new Map<string, string[]>(JSON.parse(filterMapString));
```

Comment:
For a string like:
```typescript
[["url.port",["5601","8200","9200","9292","443"]],["tags",["api"]],["observer.geo.name",["us-east-1","fairbanks"]]]
```

`JSON.parse()` will yield an array like:

```typescript
[
  ['url.port', ['5601', '8200'...]],
  ['tags', ['api']],
  ...
]
```

It felt cleaner to me to feed this array to the `Map` constructor and call `map.get(fieldName)` rather than something like `parsedArray.find(f => f[0] === fieldName)`. I can modify the code to not rely on `Map` if you think it'll be more readable.

I am noticing that we're swallowing errors and returning an empty object, I think we should actually throw an error that will get logged to the console now that I'm seeing this again, so I went ahead and added that modification in https://github.com/elastic/kibana/pull/50837/commits/3b16fc95438c4c7cf92bcccfc4234bf5c156a943.

Reply:
Oh, I didn't realize the data was shaped that way, and I didn't realize the map constructor would handle arrays that way. Makes sense!